test/unit_CartesianProduct.jl

```diff
@@ -273,3 +273,19 @@ for N in [Float64, Float32, Rational{Int}]
           EmptySet
     @test b × e == e × b == cpa × e == e × cpa == e × e == e
 end
+
+for N in [Float64, Float32]
+    #is_intersection_empty
+    i1 = Interval(N[0, 1])
+    i2 = Interval(N[2, 3])
+    h1 = Hyperrectangle(low=N[3, 4], high=N[5, 7])
+    h2 = Hyperrectangle(low=N[5, 5], high=N[6, 8])
+    cpa1 = CartesianProductArray([i1, i2, h1])
+    cpa2 = CartesianProductArray([i1, i2, h2])
+    G = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(1))])
+    G_empty = HPolyhedron([HalfSpace(N[1, 0, 0, 0], N(-1))])
+
+    @test is_intersection_empty(cpa1, cpa2) == is_intersection_empty(Approximations.overapproximate(cpa1), Approximations.overapproximate(cpa2))
+    @test is_intersection_empty(cpa1, G_empty) == is_intersection_empty(Approximations.overapproximate(cpa1), G_empty)
+    @test is_intersection_empty(cpa1, G) == is_intersection_empty(Approximations.overapproximate(cpa1), G)
```

Comment:
* Add `using LazySets.Approximations: overapproximate` to the top.
* Pull out `cpa1_box = overapproximate(cpa1)`.
* Add the expected `Bool` value.